pkg/metrics/metrics.go

```diff
@@ -211,15 +211,21 @@ func (s *HistogramWithScale) With(labelValues ...string) ScalableHistogram {
 
 // Start implements ScalableHistogram.
 func (s *HistogramWithScale) Start() {
-	s.timer = metrics.NewTimer(s.histogram)
-	s.timer.Unit(s.unit)
+	s.start = time.Now()
+}
+
+// StartAt implements ScalableHistogram.
+func (s *HistogramWithScale) StartAt(t time.Time) {
+	s.start = t
 }
 
 // ObserveDuration implements ScalableHistogram.
 func (s *HistogramWithScale) ObserveDuration() {
-	if s.timer != nil {
-		s.timer.ObserveDuration()
+	d := float64(time.Since(s.start).Nanoseconds()) / float64(s.unit)
```

Comment:
goo catch, wdyt of just returning if the unit is <= 0 to avoid generating invalid observations?

Reply:
Yes, we need at least that as a last line of defence. However, since we are already making clear to the users of the API that they should use the provided NewHistogramWithScale constructor anyway, I would also make NewHistogramWithScale return nil, errors.New("provided unit must be > 0")

Reply:
I'm reluctant to add an error throwing in there as it would have to be sent down the stack, where they never expect failures while allocating a metric type. If you are ok with it I would leave only the defensive code to avoid panic and the responsibility for correct unit value passing to the user of this Type.

Reply:
Are you referring to the call of NewHistogramWithScale in RegisterStatsd of statsd.go? If yes, I fail to see the problem: we are in control of the call there, and we know it is called with a correct value (time.Millisecond), so it would be fine to ignore the error returned by NewHistogramWithScale there, no?

Reply:
Hey. We are in a hurry to get that PR merged, so that it is included in the next release. So jsyk, I'm going to try to do the error return for NewHistogramWithScale in a commit within this PR.